### DISCUSSION TITLE
dde-session支持“会话恢复“功能讨论

### DISCUSSION THREAD

**Discussion by xzl01** [linuxdeepin/developer-center#4023] [Features Request & Ideas | 特性请求 & 头脑风暴]
会话恢复功能：指的是在用户注销或者关机，能保留注销前的状态。在用户再次登录时，自动打开相应的应用。
期望dde-session可以支持此协议
参考资料：
- X11：https://www.x.org/releases/X11R7.7/doc/libSM/xsmp.html
- wayland(WIP)：https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/18
- KDE: https://develop.kde.org/docs/features/session-managment/
- GNOME: https://lira.no-ip.org:8443/doc/gnome-session/dbus/gnome-session.html

**Comment by xzl01** [comment]
CC @meiyixiang

**Comment by meiyixiang** [comment]
相当于保留关机时或注销前的系统运行状态，是这个意思吗？内容我记录了，需要研发和我自己评估。

**Comment by meiyixiang** [comment]
你是希望仅仅只是把应用还原至注销前状态呢？还是包括关机前的软件运行或文本编辑状态

**Reply by BLumia** [reply]
取决于应用程序支持情况。应用应当在关机时有机会来存储应用状态（需要应用主动去实现相关的支持逻辑），并在开机进行恢复时根据存储的状态进行恢复。如果我们提供相关支持的话，首先会话管理应当提供这种支持，然后需要 DTK 增加相应的接口。大致上是这样的。
